.gitlab-ci.yml

```diff
@@ -143,6 +143,42 @@ default:
   - mkdir -p $HOME/.docker && echo $DOCKER_AUTH_CONFIG > $HOME/.docker/config.json
   - docker login --username $CI_REGISTRY_USER --password $CI_REGISTRY_PASSWORD $CI_REGISTRY
 
+.mantra-setup: &mantra-setup
+  - |
+    echo "Setting up Mantra test results uploader..."
+    curl -sSfL \
+      https://raw.githubusercontent.com/mendersoftware/mender-qa/074b590110e1793ada4acac7bb7b1add8fe40981/scripts/mantra_post_test_results \
+      -o /usr/local/bin/mantra_post_test_results
+    chmod +x /usr/local/bin/mantra_post_test_results
+
+.mantra-post-results: &mantra-post-results
+  - |
+    if [ -n "$MANTRA_PROJECT_NAME" ] && [ "$CI_PIPELINE_SOURCE" = "schedule" ]; then
+      # Generate MANTRA_BUILD_NAME for nightly runs only
+      MANTRA_BUILD_NAME="nightly-$(date +%Y-%m-%d)"
+
+      echo "Uploading test results to Mantra (project=$MANTRA_PROJECT_NAME, build=$MANTRA_BUILD_NAME)"
```

Comment:
If it would not be easier to use naming schema where we just add prefix / postfix to existing GitLab variable containing project / repo name?